cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/ApacheSparkContext.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.spark;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.spark.SparkContext;
+import org.apache.spark.api.java.JavaSparkContext;
+
+import java.util.Map;
+
+/**
+ * Interface to provide abstraction for accessing either scala or java version of the Apache SparkContext.
+ */
+interface ApacheSparkContext {
+
+  /**
+   * Creates a Spark RDD for the given {@link InputFormat} class.
+   *
+   * @param inputFormatClass the {@link InputFormat} class
+   * @param keyClass the key type
+   * @param valueClass the value type
+   * @param hConf the hadoop configuration
+   * @param <R> type of the resulting RDD
+   * @param <K> type of the key
+   * @param <V> type of the value
+   * @return a Spark RDD
+   */
+  <R, K, V> R createRDD(Class<? extends InputFormat> inputFormatClass,
+                        Class<K> keyClass, Class<V> valueClass, Configuration hConf);
+
+  /**
+   * Writes the RDD into dataset.
+   *
+   * @param rdd the RDD to write to dataset
+   * @param datasetName name of the dataset
+   * @param arguments arguments for the creation of the dataset
+   * @param keyClass the key type
+   * @param valueClass the value type
+   * @param hConf the hadoop configuration
+   * @param <R> type of the RDD
+   * @param <K> type of the key
+   * @param <V> type of the value
+   */
+  <R, K, V> void saveAsDataset(R rdd, String datasetName, Map<String, String> arguments,
+                               Class<K> keyClass, Class<V> valueClass, Configuration hConf);
+
+  /**
+   * Returns the Apache {@link SparkContext} or {@link JavaSparkContext}, depending on the implementation.
+   */
+  <T> T getContext();
+
+  /**
+   * Stops the Apache SparkContext.
```

Comment:
you mean the Apache {@link SparkContext} or {@link JavaSparkContext}, depending on the language used? Or does it stop this ApacheSparkContext?